### DISCUSSION TITLE
Pinterest - Search Function

### DISCUSSION THREAD

**Discussion by chiestroud** [nss-evening-cohort-14/evening-client-side#170] [Help Me!] [answered]
Your Name: Chie Stroud

Have you searched the other help tickets? 
Yes I have.

Describe your issue
I have a search bar on the navbar. I created "Promise.all" to be able to search both Boards and Pins, but only results for Boards are printed on the DOM. However, it is giving me what I want on the console. If there is a match with what I searched, it gives me the correct item back on the console. I think I am missing something or not understanding how Promise.all work, but I cannot figure out what it is.
![image](https://user-images.githubusercontent.com/68720317/110504649-ff4a4400-80c2-11eb-8c56-bc41dece35ae.png)

List any error messages that appear
No error message. Console is showing what I want to see, but not on the DOM.

What have you googled?
I am lost at what I should search at the moment. I believe I am not understanding the flow of Promise.all as much as I should so I googled how to use Promise.all.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all
https://www.freecodecamp.org/news/promise-all-in-javascript-with-example-6c8c5aea3e32/
https://stackoverflow.com/questions/36606129/promise-all-return-only-first-data

What have you tried? Tell us everything you have tried.
I changed the order to list showPin function first before showBoard function, but it still printed Board on the DOM.
I hid showBoard function. This allows me to show the searched Pin on the DOM. 
I created a showPinBoard function hoping it to return both pins and boards of the match, but I only confused myself even further.

Links to my repo
1. This is a link to my navigationEvents.js - as search bar is on my navbar.
https://github.com/chiestroud/ASSIGNMENT-Firebase-Pinterest/blob/stretch/src/javascripts/events/navigationEvent.js
2. These two below are links to my pinData & boardData where I created searchPin/searchBoard functions.
https://github.com/chiestroud/ASSIGNMENT-Firebase-Pinterest/blob/stretch/src/javascripts/helpers/data/pinData.js
https://github.com/chiestroud/ASSIGNMENT-Firebase-Pinterest/blob/stretch/src/javascripts/helpers/data/boardData.js
3. This last link is  where I tried to create "Promise.all" to have both pin/board searchable and printable to DOM at the same time.
https://github.com/chiestroud/ASSIGNMENT-Firebase-Pinterest/blob/stretch/src/javascripts/helpers/data/pinBoardsData.js

Thank you!

**Comment by drteresavasquez** [comment]
Is it possible that the DOM is being cleared by the boards? So you'd probably need to create a new view for both thing to show. Does that make sense?

**Reply by chiestroud** [reply]
Yes it makes sense. I tried to create another view to loop through both boards and pins, but I cannot wrap my head around how to loop both pin and board. If I use forEach I only succeeded in printing one. If I pass forEach((pin)) then I can print pin, and if I pass forEach((board)) then only board prints. I feel like I need to use ternary operator or if statement somewhere, but I feel like I am digging a deeper hole and cannot get out of it :( Am I going in the wrong direction?
https://github.com/chiestroud/ASSIGNMENT-Firebase-Pinterest/blob/stretch/src/javascripts/components/forms/showPinBoard.js

**Reply by drteresavasquez** [reply]
I would not modify your current code because you know it works. I would create a brand new component called searchView.js or something and loop through each of the arrays in that file using += on whichever element you're targeting. I feel like you're over thinking it. 

It's just a view. Deep breath. Step back and hit it hard! 🙂

**Reply by drteresavasquez** [reply]
Well shit! That's what you did here! 😂 let me look at this more closely. I think you may need to look at your loop.

**Reply by chiestroud** [reply]
No problem. I tried multiple different ways, but I have not had much luck so far :D

**Reply by chiestroud** [reply]
I succeeded in printing both on the DOM, but it looks really ugly and the code is not clean right now. I used two different functions to print in different places on the DOM.. I will try to refactor this and see if I can refactor and make it not this ugly.

**Reply by drteresavasquez** [reply]
But you got it on the DOM!!!!!! YAY!!!

**Reply by chiestroud** [reply]
YES! Thank you so much for the help!
Like you said, I was overthinking things by trying to use the existing functions to print them on the same spot. How I solved may not be the best way, but by creating a new function and new area to print to DOM, it solved my problem!

**Reply by drteresavasquez** [reply]
Go ahead and tell us how you were able to fix it in a separate comment so I can mark it as the answer.

**Comment by chiestroud** [comment, marked as answer]
Here is what I did for fix:
Original functions I used to search boards and pins were targeting the same div so one was overwriting the another.
I created separate two functions (files) that do not interfere with each other. One is targeting 'boardContainer' div with the destructured 'object.board' array from Promise.all and another for 'pinContainer" dive with the destructured 'object.pin' array.
